src/action/interfaces.rs

```diff
@@ -87,3 +88,38 @@ impl super::Response for Response {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+
+    use super::*;
+
+    #[test]
+    fn test_loopback_presence() {
+        let mut session = session::test::Fake::new();
+        assert!(handle(&mut session, ()).is_ok());
+
+        let mut is_loopback_present = false;
+        for i in 0..session.reply_count() {
+            let interface = &session.reply::<Response>(i).interface;
+            is_loopback_present |= interface.is_loopback();
+        }
+        assert!(is_loopback_present);
+    }
+
+    #[test]
+    fn test_response_uniqueness() {
+        use std::collections::hash_set::HashSet;
+
+        let mut session = session::test::Fake::new();
+        assert!(handle(&mut session, ()).is_ok());
+
+        let mut responses =
+            HashSet::<&Response>::with_capacity(session.reply_count());
+        for i in 0..session.reply_count() {
+            let response = &session.reply::<Response>(i);
+            assert!(!responses.contains(response));
+            responses.insert(response);
+        }
+    }
```

Comment:
In my opinion, this test does not provide you with much value. I think it should be removed (along with deriving some questionable traits for the `Response` type). The loopback test looks alright and I think its all we need.

Reply:
Okay. Do you deem `Debug` trait questionable here?

Reply:
I think `Debug` is harmless and can be safely included in most of the structs.